### PR TITLE
RA-1304 - Added REST endpoint for Data Integrity Results

### PR DESCRIPTION
**Pull request by tmarzeion**
I cant run server after I've added RestController.
@rkorytkowski I'll talk to You about that on standup tomorrow

### CODE REVIEW

File: omod/src/main/java/org/openmrs/module/dataintegrity/web/rest/controller/DataIntegrityRestSubController.java [hunk 1]
```diff
@@ -0,0 +1,18 @@
+package org.openmrs.module.dataintegrity.web.rest.controller;
```
Comment (rkorytkowski) [inline omod/src/main/java/org/openmrs/module/dataintegrity/web/rest/controller/DataIntegrityRestSubController.java]: This class shouldn't be needed.... Try removing it and see if it fixes the issue.

**Comment by tmarzeion** [comment]
@rkorytkowski I've tried running it without SubController yesterday and it didn't work :/ (But Ill remove it anyway since its redundant)
Here are server logs:
https://issues.openmrs.org/secure/attachment/49002/LOGI_AIJAR.txt

**Comment by rkorytkowski** [comment]
Looks like a number of issues:
1) port 8080 already in use
2) too old version of webservices.rest
3) edit config.xml to include require_module webservices.rest as in here https://github.com/openmrs/openmrs-module-openconceptlab/blob/master/omod/src/main/resources/config.xml#L17-L19

**Comment by tmarzeion** [comment]
@rkorytkowski 
1. Aaah, I forgot to add -Dport=8081 when running server for logs
2. I had SNAPSHOT version before and it didn't work either
3. This could be it, ill check it out

**Comment by tmarzeion** [comment]
Thx @AdamGrzybkowski for help! You are Über haxor :smile:

**Comment by tmarzeion** [comment]
@AdamGrzybkowski @rkorytkowski 
I am sorry for bothering You again, but I were trying to add SearchHandler for REST resource but somehow I am unable to make it work :/
I've added new commit:
https://github.com/openmrs/openmrs-module-dataintegrity/pull/14/commits/4cec7b29f63caaf6c08bed952e3ee29c063ec888
Here is example of how it should be done:
https://github.com/openmrs/openmrs-module-registrationcore/commit/97e1ab9e74d29d2d2ddfa5ae124f5183ce3a6097
Any thoughts?

File: omod/src/main/java/org/openmrs/module/dataintegrity/web/rest/search/ResultsByPatientSearchHandler.java [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+
+import java.util.Arrays;
+import java.util.List;
+
+@Component
+public class ResultsByPatientSearchHandler implements SearchHandler {
+
+    private static final String PATIENT = "patient";
+
+    private final SearchConfig searchConfig = new SearchConfig("default", RestConstants.VERSION_1 + DataIntegrityRestController.DATA_INTEGRITY_REST_NAMESPACE,
```
Comment (rkorytkowski) [inline omod/src/main/java/org/openmrs/module/dataintegrity/web/rest/search/ResultsByPatientSearchHandler.java]: Change to RestConstants.VERSION_1  + DataIntegrityRestController.DATA_INTEGRITY_REST_NAMESPACE + "/integrityresults

Comment (tmarzeion) [reply omod/src/main/java/org/openmrs/module/dataintegrity/web/rest/search/ResultsByPatientSearchHandler.java]: That worked :) Thanks @rkorytkowski

**Comment by rkorytkowski** [comment]
Please add the app config to the wiki at https://wiki.openmrs.org/display/docs/Patient+Summary+Widget+Documentation

**Pull request merged**